src/backend/utils/mmgr/mcxt.c

```diff
@@ -69,7 +69,7 @@ HTAB *chunks_htable = NULL;
  * CurrentMemoryContext
  *		Default memory context for allocations.
  */
-MemoryContext CurrentMemoryContext = NULL;
+volatile MemoryContext CurrentMemoryContext = NULL;
```

Comment:
I think volatility is not enough to make it thread-safe, we need something like [mutexes](https://github.com/RekGRpth/pg_curl/blob/88109625287a83d23b9717cb8a156ff84d8ca722/pg_curl.c#L95-L156).

Reply:
Good point, but mutexes are overkill here. The proper question is why threading appeared here in signal handler. I'll investigate it.